### ISSUE TITLE
SC1014 false positive for `-ot` and `-nt`

### ISSUE THREAD

**Issue by rdebath** [koalaman/shellcheck#1737]
#### For bugs
- Rule Id: SC1014 
- My shellcheck version (`shellcheck --version` or "online"):  online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC1014)
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

Note ALSO: For POSIX `-nt` is detected by SC2039 as expected, but `-ot` is not.

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

if [ lex.yy.c -ot program.l ] ; then echo 'lex.yy.c out of date' ; fi
```

#### Here's what shellcheck currently says:
```
$ shellcheck myscript
 
Line 3:
if [ lex.yy.c -ot program.l ] ; then echo 'lex.yy.c out of date' ; fi
     ^-- SC1014: Use 'if cmd; then ..' to check exit code, or 'if [[ $(cmd) == .. ]]' to check output.

$
```

#### Here's what I wanted or expected to see:
```
$ shellcheck myscript
No issues detected!

$
```

**Comment by matthewpersico** [comment]
Interesting:

```
#!/bin/bash

if [ lex.yy.c -ot program.l ] ; then echo 'lex.yy.c out of date' ; fi
if [[ lex.yy.c -ot program.l ]] ; then echo 'lex.yy.c out of date' ; fi
if [ 'lex.yy.c' -ot 'program.l' ] ; then echo 'lex.yy.c out of date' ; fi
```

```
$ shellcheck myscript
 
Line 3:
if [ lex.yy.c -ot program.l ] ; then echo 'lex.yy.c out of date' ; fi
     ^-- SC1014: Use 'if cmd; then ..' to check exit code, or 'if [[ $(cmd) == .. ]]' to check output.
 
Line 4:
if [[ lex.yy.c -ot program.l ]] ; then echo 'lex.yy.c out of date' ; fi
      ^-- SC1014: Use 'if cmd; then ..' to check exit code, or 'if [[ $(cmd) == .. ]]' to check output.

$
```

So if the filenames are quoted, then there seems to be no issue for shellcheck. All three syntaxes are evaluated properly in bash. I guess that the parser does not see a variable or a quoted string for the first token and so assumes the rest of the tokens are arguments to a command. I would propose the fix is to check ahead and if there are only three tokens and one of them is a comparison operator, then accept as written.

**Comment by koalaman** [comment]
Oops! ShellCheck looks ahead for the classic mistake of doing `[ grep -E foo bar.txt ]` before trying to parse the expression as a test expression, since this will quickly go awry. To do this, it simply looks ahead for a variable name's worth of data (a simple, unquoted alphanumeric string) and compares it to a list of common commands. This avoids the complexity of parsing and comparing a real shell token.

In this case, that variable name's worth of information stopped at the period, and `lex` is a common command name, triggering the warning.

This has now been fixed by simply requiring the command name to be followed by a space. Thanks!
